spgl1/spgl1.py

```diff
@@ -259,13 +259,13 @@ def _norm_l1_project(x, weights, tau):
         Projected array
 
     """
-    if np.all(np.isreal(x)):
+    if not np.iscomplexobj(x):
         xproj = oneprojector(x, weights, tau)
     else:
         xa = np.abs(x)
         idx = xa < _eps
         xc = oneprojector(xa, weights, tau)
-        xc /= xa
+        xc /= xa + 1e-10
```

Comment:
Can you explain why this is necessary? (I am trying to understand how the code works)

Reply:
This is needed  because if xa has zero elements it will throw a warning. Eventually the next line will just zero out these elements anyways but I thought it would be better to avoid having a warning

Reply:
I think you should use `np.where` here. Replace lines 268-269 with
```python
xc = np.where(xa < _eps, 0, xc/xa)
```
And you can then get rid of line 266.